ArmPkg/Drivers/ArmGicDxe/GicV3/ArmGicV3Dxe.c

```diff
@@ -688,8 +688,8 @@ GicV3DxeInitialize (
   ArmGicV3SetPriorityMask (0xff);
 
   // Use combined priority drop and deactivate (EOImode == 0)
-  RegValue  = ArmGicV3GetControlRegister ();
-  RegValue &= ~(UINT64)ICC_CTLR_EOImode;
+  RegValue  = (UINT32)ArmGicV3GetControlRegister ();

```

Comment:
Isn't this function returning 64 bit value?

Reply:
Unsure since its return type is UINTN... which may default to UINT64? @os-d This may be the reason RegValue was UINT64.

Reply:
hmm, at least the implementation is returning x0, which is a 64bit value. Did you change this part because it is giving you build errors?

Reply:
Yeah, UINTN is UINT64 for ARM64. I had missed that one. I would leave RegValue as UINT64 (rename it to GicV3ControlReg or something) and create a second UINT32 that is used for the other register reads that are 32 bit returns and avoid all the casts.